### ISSUE TITLE
base_url path prefix missing from output of some theme _includes files

### ISSUE THREAD

**Issue by wbreeze** [jekyll/jekyll#6837]
- [x] I believe this to be a bug, not a question about using Jekyll.
- [x] I updated to the latest Jekyll (or) if on GitHub Pages to the latest `github-pages`
- [x] I ran `jekyll doctor` to check my configuration
- [x] I read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/
- [ ] This is a feature request.

---

- [x] I am on (or have tested on) ***macOS*** 10+
- [ ] I am on (or have tested on) ***Debian/Ubuntu*** GNU/Linux
- [ ] I am on (or have tested on) ***Fedora*** GNU/Linux
- [ ] I am on (or have tested on) ***Arch*** GNU/Linux
- [ ] I am on (or have tested on) ***Other*** GNU/Linux
- [ ] I am on (or have tested on) ***Windows*** 10+

---

- [ ] I was trying to install.
- [ ] There is a broken Plugin API.
- [ ] I had an error on GitHub Pages, and I have reproduced it locally.
- [ ] I had an error on GitHub Pages, and GitHub Support said it was a Jekyll Bug.
- [ ] I had an error on GitHub Pages and I did not test it locally.
- [ ] I was trying to build.
- [x] It was another bug.

## My Reproduction Steps

```
~/Documents/p.temp$ jekyll --version
jekyll 3.7.3
~/Documents/p.temp$ jekyll new social
...
~/Documents/p.temp$ cd social
... edits to _config.yml
~/Documents/p.temp/social$ cat _config.yml
# Welcome to Jekyll!
...
baseurl: "/wnw" # the subpath of your site, e.g. /blog
url: "http://example.com" # the base hostname & protocol for your site
...
~/Documents/p.temp/social$ jekyll serve
WARN: Unresolved specs during Gem::Specification.reset:
      eventmachine (>= 0.12.9)
      rb-fsevent (>= 0.9.4, ~> 0.9)
      rb-inotify (>= 0.9.7, ~> 0.9)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
Configuration file: /Users/dclo/Documents/p.temp/social/_config.yml
            Source: /Users/dclo/Documents/p.temp/social
       Destination: /Users/dclo/Documents/p.temp/social/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
                    done in 0.793 seconds.
 Auto-regeneration: enabled for '/Users/dclo/Documents/p.temp/social'
    Server address: http://127.0.0.1:4000/wnw/
  Server running... press ctrl-c to stop.
... visit the page ...
[2018-03-09 18:41:45] ERROR `/assets/minima-social-icons.svg' not found.
```

The `_includes/social.html` file of the `minima` theme has lines like the following:

```
{% if site.github_username%}<li>
  <a href="https://github.com/{{ site.github_username| cgi_escape | escape }}">
    <svg class="svg-icon">
      <use xlink:href="{{ '/assets/minima-social-icons.svg#github' | relative_url }}"></use>
    </svg>
    <span class="username">{{ site.github_username| escape }}</span>
  </a>
</li>{% endif %}
```
Note that it uses the `| relative_url` filter.

The html produced by running `jekyll serve` or `jekyll build` produces output like this:

```
<li>
  <a href="https://github.com/jekyll">
    <svg class="svg-icon">
      <use xlink:href="/assets/minima-social-icons.svg#github"></use>
    </svg>
    <span class="username">jekyll</span>
  </a>
</li>
```

Other output, including links within the document, correctly contains the `baseurl` prefix, for example
```
<link rel="stylesheet" href="/wnw/assets/main.css">
```

Placing a print statement in `gems/jekyll-3.7.3/lib/jekyll/url_filters.rb` shows that the `relative_url` method is never called with the string, `/assets/minima-social-icons.svg#github`.

## The Output I Wanted

```
<li>
  <a href="https://github.com/jekyll">
    <svg class="svg-icon">
      <use xlink:href="/wnw/assets/minima-social-icons.svg#github"></use>
    </svg>
    <span class="username">jekyll</span>
  </a>
</li>
```

**Comment by pathawks** [comment]
This is a Minima issue rather than a Jekyll issue, and has been resolved in [jekyll/minima#201](https://github.com/jekyll/minima/pull/201). We should cut a rew release of Minima soon to make this fix available.

**Comment by wbreeze** [comment]
Ouch. Waste of our time. I was looking at the master source for minima and **using** the gem. 🙁 Thank you for checking in.
